twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -36,46 +43,71 @@ def twitter_scrooge():
       name = 'io_bazel_rules_scala/dependency/thrift/libthrift',
       actual = '@libthrift//jar')
 
-  native.maven_jar(
-      name = "scrooge_core",
-      artifact = scala_mvn_artifact("com.twitter:scrooge-core:4.6.0"),
-      sha1 = "84b86c2e082aba6e0c780b3c76281703b891a2c8",
-      server = "twitter_scrooge_maven_server",
+  scala_jar_shas = {
+      "2.11": {
+          "util_logging": "73ddd61cedabd4dab82b30e6c52c1be6c692b063b8ba310d716ead9e3b4e9267",
+          "scrooge_core": "00351f73b555d61cfe7320ef3b1367a9641e694cfb8dfa8a733cfcf49df872e8",
+          "scrooge_generator": "0f0027e815e67985895a6f3caa137f02366ceeea4966498f34fb82cabb11dee6",
+          "util_core": "5336da4846dfc3db8ffe5ae076be1021828cfee35aa17bda9af461e203cf265c"
+      },
+      "2.12": {
+          "util_logging": "c0cba01705e9321b3444adcd4a9ce27c2acefd27e14c13b5aec2c318ce1b4fdf",
+          "scrooge_core": "02a6d7cf9fe8d872dfabd20298e4315d677748708e153d8b464fd5abac9a7430",
+          "scrooge_generator": "e7d5da1e3f0e494d3c81a26f44f3e3dc92d7efd757133de8c71758646fd5a833",
+          "util_core": "65bb92e70f95cbbfc640e54a5823a16154eac1a2631dc0211347e085aaa6ed0b"
+      },
+  }
+
+  scala_version_jar_shas = scala_jar_shas[major_version]
+
+  _scala_maven_import_external(
+      name = "io_bazel_rules_scala_scrooge_core",
+      artifact = _scala_mvn_artifact("com.twitter:scrooge-core:18.6.0",
+                                     major_version),
+      jar_sha256 = scala_version_jar_shas["scrooge_core"],
+      licenses = ["notice"],
+      server_urls = maven_servers,
   )
   native.bind(
       name = 'io_bazel_rules_scala/dependency/thrift/scrooge_core',
-      actual = '@scrooge_core//jar')
+      actual = '@io_bazel_rules_scala_scrooge_core')
 
   #scrooge-generator related dependencies
-  native.maven_jar(
-      name = "scrooge_generator",
-      artifact = scala_mvn_artifact("com.twitter:scrooge-generator:4.6.0"),
-      sha1 = "cacf72eedeb5309ca02b2d8325c587198ecaac82",
-      server = "twitter_scrooge_maven_server",
+  _scala_maven_import_external(
+      name = "io_bazel_rules_scala_scrooge_generator",
+      artifact = _scala_mvn_artifact("com.twitter:scrooge-generator:18.6.0",
+                                     major_version),
+      jar_sha256 = scala_version_jar_shas["scrooge_generator"],
+      licenses = ["notice"],
+      server_urls = maven_servers,
   )
   native.bind(
       name = 'io_bazel_rules_scala/dependency/thrift/scrooge_generator',
-      actual = '@scrooge_generator//jar')
-
-  native.maven_jar(
-      name = "util_core",
-      artifact = scala_mvn_artifact("com.twitter:util-core:6.33.0"),
-      sha1 = "bb49fa66a3ca9b7db8cd764d0b26ce498bbccc83",
-      server = "twitter_scrooge_maven_server",
+      actual = '@io_bazel_rules_scala_scrooge_generator')
+
+  _scala_maven_import_external(
+      name = "io_bazel_rules_scala_util_core",
+      artifact = _scala_mvn_artifact("com.twitter:util-core:18.6.0",
```

Comment:
@johnynek This PR has a few decent changes into how folks must consider the bindings for both the scala libs, and also the version changed in scrooge. --- should we have some sort of upgrading readme/notes with these? 
e.g. if you use bazel deps you need to change X to Y, If you use scrooge you must add the bind section , etc..

Reply:
@jhnj would you be able to make a short .md file that describes how to set up in the normal case but also how to customize to point to different versions? This often comes up as a question users have but we only have the source code to point to and it can be hard to find for new users.

Reply:
@johnynek @ianoc There is a short section in the readme about selecting scala version (`Selecting Scala version
`), I can definitely add a more thorough explanation there (or another md file) if needed.

Regarding the updated thrift version + bazel-deps change, there is currently no versioning for rules_scala, would we want that? We can ofc add upgrading notes for commits that need action from the users, e.g. `76d4ab9f855f78113ee5990a84b0ad55d2417e4a requires you to change X if you're using bazel-deps` but I think that could quickly become noisy + it's not very readable.

(sidenote: the bazel-deps change actually came from an earlier PR)

Reply:
Some users don't use our provided scala_repositories, and instead wire them up from jars found by say bazel-deps. It would be good to add a section on using these rules with bazel-deps (either here, or in the bazel-deps repo).

Reply:
I think referencing commits is fine. I don't think we have the energy to do a more formal versioning for now.

Reply:
If the users just bind their own jars, basically replacing this:
``` 
native.bind(
      name = "io_bazel_rules_scala/dependency/scalatest/scalatest",
      actual = "@io_bazel_rules_scala//scala/scalatest:scalatest")
```
nothing should have changed in this PR, the name it's bound to hasn't changed.

There is a example on using jars from `scala_repositories` with bazel-deps in the bazel-deps readme but can add something here as well

Reply:
This for instance we need to call out would silently change the version of the scrooge generator used by people. It wasn't a custom binding before hand. Thats an important change we would want to call out since it would lead to strange/confusing errors. 

(in general changing the default version of the libraries is probably something we want to call out pretty loudly. could easily cause binary incompatibilities or other issues for folks even if it compiles inside their repo)

Reply:
Agree, adding notes about that 👍